include/mbgl/map/map.hpp

```diff
@@ -34,6 +34,7 @@ class Map : private util::noncopyable {
     explicit Map(View&, FileSource&,
                  MapMode mapMode = MapMode::Continuous,
                  GLContextMode contextMode = GLContextMode::Unique,
+                 ViewportMode viewportMode = ViewportMode::Default,
```

Comment:
On a second thought - for the sake of adding new features at the tail - I'm going to add this after `ConstrainMode` 👍